pkg/cliplugins/workspace/plugin/kubeconfig_test.go

```diff
@@ -701,6 +702,204 @@ func TestUse(t *testing.T) {
+				Clusters: map[string]*clientcmdapi.Cluster{
+					"workspace.kcp.dev/current": {Server: "https://test/clusters/root:foo:bar"},
+					"root:foo:bar":              {Server: "https://test/clusters/root:foo:bar"},
+				},
+				AuthInfos: map[string]*clientcmdapi.AuthInfo{"test": {Token: "test"}},
+			},
+			param: "",
+			expected: &clientcmdapi.Config{CurrentContext: "root:foo:bar",
+				Contexts: map[string]*clientcmdapi.Context{
+					"workspace.kcp.dev/current": {Cluster: "workspace.kcp.dev/current", AuthInfo: "test"},
+					"root:foo:bar":              {Cluster: "root:foo:bar", AuthInfo: "test"},
+				},
+				Clusters: map[string]*clientcmdapi.Cluster{
+					"workspace.kcp.dev/current": {Server: "https://test/clusters/root:foo:bar"},
+					"root:foo:bar":              {Server: "https://test/clusters/root:foo:bar"},
+				},
+				AuthInfos: map[string]*clientcmdapi.AuthInfo{"test": {Token: "test"}},
+			},
+			wantStdout: []string{"Updated context \"root:foo:bar\"."},
+		},
+		{
+			name: "current, no arg, overrides don't apply",
+			config: clientcmdapi.Config{CurrentContext: "workspace.kcp.dev/current",
+				Contexts:  map[string]*clientcmdapi.Context{"workspace.kcp.dev/current": {Cluster: "workspace.kcp.dev/current", AuthInfo: "test"}},
+				Clusters:  map[string]*clientcmdapi.Cluster{"workspace.kcp.dev/current": {Server: "https://test/clusters/root:foo:bar"}},
+				AuthInfos: map[string]*clientcmdapi.AuthInfo{"test": {Token: "test"}},
+			},
+			overrides: &clientcmd.ConfigOverrides{
+				AuthInfo: clientcmdapi.AuthInfo{
+					Token: "new-token",
+				},
+			},
+			param: "",
+			expected: &clientcmdapi.Config{CurrentContext: "root:foo:bar",
+				Contexts: map[string]*clientcmdapi.Context{
+					"workspace.kcp.dev/current": {Cluster: "workspace.kcp.dev/current", AuthInfo: "test"},
+					"root:foo:bar":              {Cluster: "root:foo:bar", AuthInfo: "test"},
+				},
+				Clusters: map[string]*clientcmdapi.Cluster{
+					"workspace.kcp.dev/current": {Server: "https://test/clusters/root:foo:bar"},
+					"root:foo:bar":              {Server: "https://test/clusters/root:foo:bar"},
+				},
+				AuthInfos: map[string]*clientcmdapi.AuthInfo{"test": {Token: "test"}},
+			},
+			wantStdout: []string{"Created context \"root:foo:bar\" and switched to it."},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var got *clientcmdapi.Config
+
+			clusterName := tt.config.Contexts[tt.config.CurrentContext].Cluster
+			cluster := tt.config.Clusters[clusterName]
+			u := parseURLOrDie(cluster.Server)
+			u.Path = ""
+
+			streams, _, stdout, stderr := genericclioptions.NewTestIOStreams()
+
+			kc := &KubeConfig{
+				startingConfig: tt.config.DeepCopy(),
+				currentContext: tt.config.CurrentContext,
+				overrides:      tt.overrides,
+
+				modifyConfig: func(config *clientcmdapi.Config) error {
+					got = config
+					return nil
+				},
+				IOStreams: streams,
+			}
+			err := kc.CreateContext(context.TODO(), tt.param)
```

Comment:
`context.Background()` is probably better here, as `TODO()` is more for when you need a context, but don't have a way to pass one in (i.e. you intend to come back and plumb it).

Reply:
done